tests/integration/test_dataapi.py

```diff
@@ -59,30 +52,100 @@ def _build_base_url(self):
 
         return base_url
 
-
     def test_real_request(self):
         """Make a request against Data Api to ensure the SDK works"""
         client = DataApi()
-        res = client.request(self._build_base_url() + endpoint, security, consumer_secret, request,
+
+        test_url = self._build_base_url() + items_endpoint
+
+        print("test_url is", test_url)
+
+        res = client.request(test_url, security, consumer_secret, items_request,
                              action)
-        print(res.request.url)
-        print(res.request.body)
-        print(res.content)
+
         returned_json = res.json()
+
         assert len(returned_json['data']) > 0
+
         returned_ref = returned_json['data'][0]['reference']
-        assert returned_ref in request['references']
+
+        assert returned_ref in items_request['references']
 
     def test_paging(self):
         """Verify that paging works"""
         client = DataApi()
-        pages = client.request_iter(self._build_base_url() + endpoint, security, consumer_secret,
-                                    request, action)
+
+        test_url = self._build_base_url() + items_endpoint
+
+        print("test_url is", test_url)
+
+        pages = client.request_iter(test_url, security, consumer_secret,
+                                    items_request, action)
+
         results = set()
+
         for page in pages:
             if page['data']:
                 results.add(page['data'][0]['reference'])
 
         assert len(results) == 2
         assert results == {'item_2', 'item_3'}
 
+    def test_real_question_request(self):
+        """Make a request against Data Api to ensure the SDK works"""
+        client = DataApi()
+
+        test_url = self._build_base_url() + questions_endpoint
+
+        print("test_url is", test_url)
+
+        questions_request['limit'] = 3
+
+        res = client.request(test_url, security, consumer_secret, questions_request,
+                             action)
+
+        returned_json = res.json()
+
+        assert len(returned_json['data']) > 0
+
+        keys = set()
+
+        for key, value in self.__iteritems(returned_json['data']):
+            keys.add(key)
+
+        assert keys == {'py-sdk-test-2019-1', 'py-sdk-test-2019-2'}
+
+    def test_question_paging(self):
+        """Verify that paging works"""
+        client = DataApi()
+
+        test_url = self._build_base_url() + questions_endpoint
+
+        print("test_url is", test_url)
+
+        pages = client.request_iter(test_url, security, consumer_secret,
+                                    questions_request, action)
+
+        results = []
+
+        for page in pages:
+            if page['data']:
+                results.append(page['data'])
+
+        keys = set()
+
+        for row in results:
+            for key in row.keys():
+                keys.add(key)
+
+        assert len(results) == 3
```

Comment:
What's the cause of `results` having a length of 3, while there's only 2 references checked below for keys? Is there an empty page for some reason that we should try to clear out?

For the items endpoint (an array), the length is 2 instead of 3 which matches the number of keys.

Reply:
With this difference, one of the test items has 2 questions. As a result it returns 3 different question objects, each one with the appropriate key. Meanwhile the other API test calls the API directly then parses the JSON manually - so it finds 2 items, one with 2 questions one with 3.

Reply:
Does that make sense or did I mangle my explanation?